crates/miden-objects/src/block/proven_block.rs

```diff
@@ -1,55 +1,23 @@
-use alloc::vec::Vec;
-
-use crate::block::{
-    BlockAccountUpdate,
-    BlockHeader,
-    BlockNoteIndex,
-    BlockNoteTree,
-    OutputNoteBatch,
-};
-use crate::note::Nullifier;
-use crate::transaction::{OrderedTransactionHeaders, OutputNote};
+use crate::MIN_PROOF_SECURITY_LEVEL;
+use crate::block::{BlockBody, BlockHeader, BlockProof, SignedBlock};
 use crate::utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
-use crate::{MIN_PROOF_SECURITY_LEVEL, Word};
 
 // PROVEN BLOCK
 // ================================================================================================
 
-/// A block in the Miden chain.
+/// Represents a block in the Miden blockchain that has been signed and proven.
 ///
-/// A block is built from batches of transactions, i.e. multiple
-/// [`ProvenBatch`](crate::batch::ProvenBatch)es, and each batch contains multiple
-/// [`ProvenTransaction`](crate::transaction::ProvenTransaction)s.
+/// Blocks transition from proposed, signed, and proven states. This struct represents the final,
+/// proven state of a block.
 ///
-/// It consists of the following components:
-/// - A [`BlockHeader`] committing to the current state of the chain and against which account, note
-///   or nullifier inclusion or absence can be proven. See its documentation for details on what it
-///   commits to. Eventually, it will also contain a ZK proof of the validity of the block.
-/// - A list of account updates for all accounts updated in this block. For private accounts, the
-///   update contains only the new account state commitments while for public accounts, the update
-///   also includes the delta which can be applied to the previous account state to get the new
-///   account state.
-/// - A list of new notes created in this block. For private notes, the block contains only note IDs
-///   and note metadata while for public notes the full note details are included.
-/// - A list of new nullifiers created for all notes that were consumed in the block.
-/// - A list of transaction headers that were included in the block.
+/// Proven blocks are the final, canonical blocks in the chain.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct ProvenBlock {
-    /// The header of the block, committing to the current state of the chain.
-    header: BlockHeader,
-
-    /// Account updates for the block.
-    updated_accounts: Vec<BlockAccountUpdate>,
-
-    /// Note batches created by the transactions in this block.
-    output_note_batches: Vec<OutputNoteBatch>,
-
-    /// Nullifiers created by the transactions in this block through the consumption of notes.
-    created_nullifiers: Vec<Nullifier>,
+    /// The signed block that the [`BlockProof`] is based on.
+    signed_block: SignedBlock,
```

Comment:
I would not use `SignedBlock` here but rather just have `header` and `body` separately:

```rust
pub struct ProvenBlock {
    header: BlockHeader,
    body: BlockBody,
    proof: BlockProof,
}
```